cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/LoadPersistedConcepts.kt

```diff
@@ -0,0 +1,318 @@
+        val loc =
+            try {
+                val regex =
+                    Regex(
+                        "(?<startLine>\\d+):(?<startColumn>\\d+)-(?<endLine>\\d+):(?<endColumn>\\d+)"
+                    )
+                val region =
+                    regex.matchEntire(location.region)
+                        ?: throw IllegalArgumentException(
+                            "Invalid region format: \${location.region}"
+                        )
+                val startLine =
+                    region.groups["startLine"]?.value?.toIntOrNull()
+                        ?: throw IllegalArgumentException("Invalid or missing startLine in region.")
+                val startColumn =
+                    region.groups["startColumn"]?.value?.toIntOrNull()
+                        ?: throw IllegalArgumentException(
+                            "Invalid or missing startColumn in region."
+                        )
+                val endLine =
+                    region.groups["endLine"]?.value?.toIntOrNull()
+                        ?: throw IllegalArgumentException("Invalid or missing endLine in region.")
+                val endColumn =
+                    region.groups["endColumn"]?.value?.toIntOrNull()
+                        ?: throw IllegalArgumentException("Invalid or missing endColumn in region.")
+
+                PhysicalLocation(
+                    uri = URI(location.file),
+                    region = Region(startLine, startColumn, endLine, endColumn),
+                )
+            } catch (ex: Exception) {
+                log.error("Failed to parse the location: ${location.region}: ${ex.message}", ex)
+                return emptyList()
+            }
+
+        // find the matching node
+        return translationResult.getNodesByRegion(location = loc, clsName = location.type).also {
+            nodes ->
+            if (nodes.size != 1) {
+                logger.warn(
+                    "Found ${nodes.size} nodes for location $loc. Expected exactly one node."
+                )
+            }
+        }
+    }
+
+    /**
+     * This is a helper function to add a [Concept] node to the [underlyingNode] as described in the
+     * provided [ConceptEntry].
+     *
+     * @param underlyingNode The node to which the concept will be added.
+     * @param concept The [ConceptEntry] containing the concept information.
+     */
+    private fun addConcept(underlyingNode: Node, concept: ConceptEntry) {
+        logger.debug("Adding concept {} to node {}.", concept, underlyingNode)
+        underlyingNode.conceptBuildHelper(
+            name = concept.name,
+            underlyingNode = underlyingNode,
+            constructorArguments =
+                concept.constructorArguments?.associate { arg -> arg.name to arg.value }
+                    ?: emptyMap(),
+            connectDFGUnderlyingNodeToConcept =
+                concept.dfg?.fromThisNodeToConcept ?: false, // TODO: this `?: false` is not nice
```

Comment:
Remove the TODO. Can't we use a default value for the dfg entries if nothing is provided by the user?